src/gui/speedlimitdialog.cpp

```diff
@@ -50,57 +62,110 @@ SpeedLimitDialog::~SpeedLimitDialog()
     delete m_ui;
 }
 
-// -2: if cancel
-long SpeedLimitDialog::askSpeedLimit(QWidget *parent, bool *ok, const QString &title, const long defaultVal, const long maxVal)
+void SpeedLimitDialog::askNewSpeedLimits(QWidget *parent, bool *ok, const QString &title, SpeedLimits &speedLimits)
```

Comment:
It should be: `bool SpeedLimitDialog::askNewSpeedLimits(QWidget *parent, const QString &title, SpeedLimits &speedLimits)`

Reply:
In fact it shouldn't be a blocking action (regardless of the returning value/type), it should be returning the results via signal.

Reply:
I don't understand what i would do with the bool. The second comment if you mean the way it's done in updownrationdialog then that doesn't seem trivial at all (to me).

Reply:
>I don't understand what i would do with the bool. 

you use `return true/false;` to return the result of executing this function instead of assigning the results to `ok` variable.

>The second comment if you mean the way it's done in updownrationdialog then that doesn't seem trivial at all (to me).

oh well, it is optional.

Reply:
bool done.